hrms/payroll/doctype/salary_structure/salary_structure.json

```diff
@@ -73,7 +73,7 @@
   },
   {
    "default": "Monthly",
-   "depends_on": "eval:(!doc.salary_slip_based_on_timesheet)",
+   "depends_on": "eval:doc.salary_slip_based_on_timesheet == 0",
```

Comment:
The commits don't match the title of the PR. It says do not hide payroll frequency but is actually doing it and just resetting frequency if the salary slip is based on the timesheet right?

Can we keep clearer titles and commits? It shows up in the changelog for every version release so just pointing out.